### ISSUE TITLE
Convert Deprecated sys.sysprocesses to sys.dm_exec_sessions 

### ISSUE THREAD

**Issue by sparkledoney** [amachanic/sp_whoisactive#57]
sys.sysprocesses is deprecated and so it is causing "complaints" in our database project as it cannot resolve the reference: SQL71502

**Comment by erikdarlingdata** [comment]
The problem is that `dm_exec_sessions` is missing columns in older (and, yeah, unsupported) versions of SQL Server that make replacing it impossible. I had to revert back to `sysprocesses` in a couple cases ([1](https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/issues/2961), [2](https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/issues/2962)) where it was breaking other scripts I work on.

**Comment by amachanic** [comment]
I started a migration project several years ago. It's doable, but way more difficult than you might think. The big issue for me wasn't missing columns, but rather performance. sysprocesses is basically equivalent to sys.dm_exec_sessions joined to sys.dm_exec_requests joined to sys.dm_os_waiting_tasks -- but it handles all of the "join" logic internally, direct from memory, in one shot. It's much, much faster to query that one view than to hit those three and let the query processor do its thing.

Adding extra time there means taking time from somewhere else. sp_whoisactive uses sysprocesses as its core, and then joins to all of those other tables mentioned above. Perhaps we could figure out some fancy SQL maneuvers so that we could only hit each table exactly once, but I didn't quite get there. And is sysprocesses **really** going anywhere? It's been "deprecated" for 16 years now...

Bottom line: If anyone wants to take a stab at fixing this, go for it, and submit a PR! But I think a much nicer option for the foreseeable future is to simply disable the warning :-)
